components/edit_post/edit_post.tsx

```diff
@@ -0,0 +1,481 @@
+
+        return () => {
+            document.removeEventListener('paste', handlePaste);
+        };
+    });
+
+    const isSaveDisabled = () => {
+        const {post} = editingPost;
+        const hasAttachments = post && post.file_ids && post.file_ids.length > 0;
+
+        if (hasAttachments) {
+            return !rest.canEditPost;
+        }
+
+        if (editText.trim() !== '') {
+            return !rest.canEditPost;
+        }
+
+        return !rest.canDeletePost;
+    };
+
+    const applyHotkeyMarkdown = (e: React.KeyboardEvent) => {
+        const res = Utils.applyHotkeyMarkdown(e);
+
+        setEditText(res.message);
+        if (textboxRef.current) {
+            Utils.setSelectionRange(
+                textboxRef.current.getInputBox(),
+                res.selectionStart,
+                res.selectionEnd,
+            );
+        }
+    };
+
+    const handleRefocusAndExit = (refocusId: string|null) => {
+        if (refocusId) {
+            setTimeout(() => {
+                const element = document.getElementById(refocusId);
+                if (element) {
+                    element.focus();
+                }
+            });
+        }
+
+        setCaretPosition(0);
+        setPostError(null);
+        setErrorClass('');
+        setShowEmojiPicker(false);
+        setRenderScrollbar(false);
+    };
+
+    const handleHide = (doRefocus = true) => {
+        setEditText(editingPost.post?.message || '');
+        actions.unsetEditingPost();
+        handleRefocusAndExit(doRefocus && editingPost.refocusId ? editingPost.refocusId : null);
+    };
+
+    const handleEdit = async () => {
+        if (!editingPost.post || isSaveDisabled()) {
+            return;
+        }
+
+        const updatedPost = {
+            message: editText,
+            id: editingPost.postId,
+            channel_id: editingPost.post.channel_id,
+        };
+
+        if (postError) {
+            setErrorClass('animation--highlight');
+            setTimeout(() => setErrorClass(''), Constants.ANIMATION_TIMEOUT);
+            return;
+        }
+
+        if (updatedPost.message === (editingPost.post?.message_source || editingPost.post?.message)) {
+            handleHide();
+            return;
+        }
+
+        const hasAttachment = Boolean(
+            editingPost.post?.file_ids && editingPost.post?.file_ids.length > 0,
+        );
+        if (updatedPost.message.trim().length === 0 && !hasAttachment) {
+            handleHide(false);
+
+            const deletePostModalData = {
+                ModalId: ModalIdentifiers.DELETE_POST,
+                dialogType: DeletePostModal,
+                dialogProps: {
+                    post: editingPost.post,
+                    isRHS: editingPost.isRHS,
+                },
+            };
+
+            actions.openModal(deletePostModalData);
+            return;
+        }
+
+        actions.addMessageIntoHistory(updatedPost.message);
```

Comment:
Is this desirable? The edited text to become part of the history of messages?

Reply:
good question  .... I am not 100% sure what the history is used for, so maybe not, since the original message is already in there

Reply:
This history is for showing different inputs when you hit `ctrl+up`. It is really useful when you are typing commands.

Reply:
well ... first of all: thanks for clarifying that. ... but I cannot use that hotkey on my machine (on localhost and community), since it does nothing. Is it still in place?